<launch>
<!--    arg name    "pcd_data" 
        input data  "cloud_in" 
        frame ID    "map_tf"       -->

  <node pkg="nodelet" type="nodelet" name="pcl_manager" args="manager" output="screen" />

  <arg name="pcd_data" default="/home/ymp/catkin_ws/src/pcd/bulc_cloud.pcd"/>

  <node name="pcd_to_pointcloud" pkg="pcl_ros" type="pcd_to_pointcloud"  args="$(arg pcd_data) 0.1"> 
    <param name="frame_id" value="map"/> 
    <remap from="cloud_pcd" to="/cloud_in" /> 
  </node>
  <node pkg="tf" type="static_transform_publisher" name="map_tf" args="0 0 0 0 0 3.14 /cloud_in map_tf 100" />

  <node name="plane_segment"  pkg="pcl_something" type="planeDetect"/>

  <node pkg="nodelet" type="nodelet" name="passthrough_z" args="load pcl/PassThrough pcl_manager" output="screen">
    <remap from="~input" to="/cloud_seg" />
  <remap from="~output" to="/cloud_seg_z" />
    <rosparam>
      filter_field_name: z
      filter_limit_min: -10.0
      filter_limit_max: 10.0
      filter_limit_negative: False
    </rosparam>
  </node>

  <node name="rviz" pkg="rviz" type="rviz" />

</launch>